src/sentry/static/sentry/app/views/organizationGroupDetails/actions/deleteAction.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {ModalRenderProps, openModal} from 'app/actionCreators/modal';
+import Feature from 'app/components/acl/feature';
+import FeatureDisabled from 'app/components/acl/featureDisabled';
+import Button from 'app/components/button';
+import DropdownLink from 'app/components/dropdownLink';
+import LinkWithConfirmation from 'app/components/links/linkWithConfirmation';
+import MenuItem from 'app/components/menuItem';
+import {IconDelete} from 'app/icons';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {Organization, Project} from 'app/types';
+import {analytics} from 'app/utils/analytics';
+
+type Props = {
+  organization: Organization;
+  project: Project;
+  onDelete: () => void;
+  onDiscard: () => void;
+};
+
+function DeleteAction({project, organization, onDiscard, onDelete}: Props) {
+  function renderDiscardDisabled({children, ...props}) {
+    return children({
+      ...props,
+      renderDisabled: ({features}: {features: string[]}) => (
+        <FeatureDisabled alert featureName="Discard and Delete" features={features} />
+      ),
+    });
+  }
+
+  function renderDiscardModal({Body, closeModal}: ModalRenderProps) {
+    return (
+      <Feature
+        features={['projects:discard-groups']}
+        hookName="feature-disabled:discard-groups"
+        organization={organization}
+        project={project}
+        renderDisabled={renderDiscardDisabled}
+      >
+        {({hasFeature, renderDisabled, ...props}) => (
+          <React.Fragment>
+            <Body>
+              {!hasFeature &&
+                typeof renderDisabled === 'function' &&
+                renderDisabled({...props, hasFeature, children: null})}
+              {t(
+                `Discarding this event will result in the deletion of most data associated with this issue and future events being discarded before reaching your stream. Are you sure you wish to continue?`
+              )}
+            </Body>
+            <div className="modal-footer">
```

Comment:
Can we use Footer that's part of ModalRenderProps instead?

Reply:
I will check this, but I prefer to create a new PR with the updates 😉